graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -10,14 +10,11 @@
     PROPERTY_TYPE_LINK_ID, PropertyDescriptor, parse_default_property_value,
     validate_supported_property_type_id
 )
+from .utils import toposort_classes
 
 
 def _validate_edge_has_defined_endpoint_types(class_name, properties):
     """Validate that the edge properties dict has defined types for the in/out link properties."""
-    exception_classes = frozenset({ORIENTDB_BASE_EDGE_CLASS_NAME})
-    if class_name in exception_classes:
-        return
```

Comment:
The base edge class is guaranteed to be abstract and so we'll never get here. Also see bottom comment. 
            if class_name in orientdb_base_classes:
                # Special-case the V and E base classes:
                # OrientDB won't let us make them abstract, but we don't want to create
                # any vertices or edges with those types either.
                # Pretend they are marked abstract in OrientDB's schema.
                abstract = True

Reply:
Document in the name of the function the precondition that the class is abstract

Reply:
@pmantica1, I am a bit worried about this change and how the GraphQL schema generation code will handle it. The case I am worried about is this: the user has defined an abstract edge class in OrientDB, and has only defined the type of *one* of the edge's endpoints.

This is not a case that happens in our internal use case, but since this code is open-source, it may occur in someone else's use case. Would you mind adding a test (in a new PR) for the schema generation code that attempts to pin down what happens in such a situation? What should happen in that situation?

I don't think you need to actually have that schema exist in OrientDB -- you should be able to test for it by constructing a simple OrientDB schema list-of-dicts response by hand.

Reply:
Predrag I think you might be reviewing an intermediate commit. I renamed this method to _validate_non_abstract_edge_has_defined_endpoint_types in the last commit of the pull request and I only run this validation if the edge is not abstract. So nothing should happen in the use case you mentioned?